utils/custom.js

```diff
@@ -115,6 +124,7 @@ const handleAnonymousData = async (data, req, postAuthorId, myAnonymousId, anony
           ...child,
           latest_children: {comment: childCommentLv2},
           user_id: null,
+          karmaScores: roundingKarmaScore(user?.karma_score || 0),
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code assumes that `user.karma_score` will always be a number or undefined, but there's no validation to confirm this. If `user.karma_score` is not a number, the `roundingKarmaScore` function might behave unexpectedly. Consider adding a type check before calling `roundingKarmaScore`.

```diff
- karmaScores: roundingKarmaScore(user?.karma_score || 0),
+ karmaScores: typeof user?.karma_score === 'number' ? roundingKarmaScore(user.karma_score) : 0,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
the validation handled in roundingKarmaScore function